crates/pink/sidevm/sidevm/src/net.rs

```diff
@@ -114,16 +129,123 @@ mod impl_hyper {
             self: Pin<&mut Self>,
             cx: &mut Context<'_>,
         ) -> Poll<Option<Result<Self::Conn, Self::Error>>> {
-            let mut accept = Acceptor {
-                listener: self.get_mut(),
-            };
-            match Pin::new(&mut accept).poll(cx) {
-                Poll::Ready(Ok((conn, _addr))) => Poll::Ready(Some(Ok(conn))),
-                Poll::Ready(Err(err)) => Poll::Ready(Some(Err(err))),
-                Poll::Pending => Poll::Pending,
-            }
+            let (conn, _addr) = ready_ok!(Pin::new(&mut self.accept()).poll(cx));
+            Poll::Ready(Some(Ok(conn)))
         }
     }
+
+    impl TcpListener {
+        /// Convert the listener into another one that outputs AddrStreams.
+        pub fn into_addr_incoming(self) -> AddrIncoming {
+            AddrIncoming { listener: self }
+        }
+    }
+
+    /// A wrapper of  TcpListener that outputs AddrStreams.
+    pub struct AddrIncoming {
```

Comment:
Does `AddrStream` mean the tcp stream and the incoming address?

Reply:
It means the type `struct AddrStream` which, yes, contains the TCP stream and the remote address.